src/dipdup/index.py

```diff
@@ -405,13 +415,19 @@ async def _on_match(
 
         if matched_big_map.action.has_key:
             key_type = handler_config.key_type_cls
-            key = key_type.parse_obj(matched_big_map.key)
+            try:
+                key = key_type.parse_obj(matched_big_map.key)
+            except ValidationError:
+                raise InvalidDataError(matched_big_map.key, key_type)
         else:
             key = None
 
         if matched_big_map.action.has_value:
             value_type = handler_config.value_type_cls
-            value = value_type.parse_obj(matched_big_map.value)
+            try:
+                value = value_type.parse_obj(matched_big_map.value)
+            except ValidationError:
```

Comment:
`except Error as e raise Error from e` as first exception is a direct cause of second one.